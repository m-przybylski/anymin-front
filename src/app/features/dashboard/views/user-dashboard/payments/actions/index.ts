import * as PaymentsApiActions from './payments-api.actions';
import * as PaymentsInitActions from './payments-init.actions';

export { PaymentsInitActions, PaymentsApiActions };
