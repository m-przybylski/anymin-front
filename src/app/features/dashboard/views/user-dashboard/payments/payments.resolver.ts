import { Resolve } from '@angular/router';
import { GetPayoutMethod, PayoutsService } from '@anymind-ng/api';
import { Observable, of } from 'rxjs';
import { Logger } from '@platform/core/logger';
import { LoggerFactory, LoggerService } from '@anymind-ng/core';
import { Injectable } from '@angular/core';
import { catchError } from 'rxjs/operators';

@Injectable()
export class PaymentsResolver extends Logger implements Resolve<GetPayoutMethod> {
  protected loggerService: LoggerService;

  constructor(private payoutsService: PayoutsService, loggerFactory: LoggerFactory) {
    super(loggerFactory.createLoggerService('PaymentsResolver'));
  }

  public resolve = (): Observable<GetPayoutMethod> =>
    this.payoutsService.getPayoutMethodsRoute().pipe(catchError(() => of({})));
}
