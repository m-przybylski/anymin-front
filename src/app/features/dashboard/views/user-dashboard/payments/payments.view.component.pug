.payments
  .payments__header
    span {{ 'DASHBOARD.PAYMENTS.PAYMENTS_METHOD.TITLE' | translate }}

  form.payments__container.payments__container--list([formGroup]="paymentsCardFormGroup")
    ul.payments-method-list(*ngIf="!(isPending$ | async); else loader")
      li.payments-method-list__item(*ngFor="let card of (paymentsCardList$ | async); let i = index")
        plat-select-payment-method-card-radio(
          [form]="paymentsCardFormGroup",
          [controlName]="paymentsCardControlName",
          [expiryDate]="card.expiryDate",
          [maskedNumber]="card.maskedNumber",
          [id]="card.id",
          [selectedCardId]="currentPaymentMethodId",
          [cardType]="card.cardType",
          (onSelectCard)="onSelectCard($event)",
          (onDeleteCard)="onDeleteCard($event)",
        )
      li.payments-method-list__item.payments-method-list__item--promocode(*ngFor="let promoCode of promoCodeList$ | async; let i = index")
        plat-select-promo-code-radio(
          [form]="paymentsCardFormGroup",
          [controlName]="paymentsCardControlName",
          [expiryDate]="promoCode.expiresAt",
          [id]="promoCode.id",
          [selectedCardId]="currentPaymentMethodId",
          [promoCodeAmount]="promoCode.amount",
          (onSelectCard)="onSelectPromoCode($event)",
        )

    ng-template(#loader)
      ul.settings-list-loader
        li.settings-list-loader__item
          plat-list-item-loader
        li.settings-list-loader__item
          plat-list-item-loader
        li.settings-list-loader__item

  .payments__container.payments__container--method
    .payments__methods
      .payments__type.payments__type--card
        div
          a.button.button--add((click)="onAddPaymentCard()")
            span(plat-flat-fab color='primary')
              span(plat-icon iconClass='add')
            span.button__text {{ 'DASHBOARD.PAYMENTS.PAYMENTS_METHOD.ADD_CARD' | translate }}

      .payments__type.payments__type--promo-code
        div
          a.button.button--add((click)="onAddPromoCode()")
            span(plat-flat-fab color='primary')
              span(plat-icon iconClass='add')
            span.button__text {{ 'DASHBOARD.PAYMENTS.PAYMENTS_METHOD.ADD_PROMO_CODE' | translate }}

  .payments__header
    span {{ 'DASHBOARD.PAYMENTS.TITLE' | translate }}

  .payments__container
    plat-setting-option(
    [title]="'DASHBOARD.PAYMENTS.INVOICE.TITLE'",
    [description]="'DASHBOARD.PAYMENTS.INVOICE.DESCRIPTION'"
    )

      button.button-mobile(plat-stroked-button, (click)="openInvoiceModal()")
        span {{ 'DASHBOARD.PAYMENTS.BUTTON.CHANGE' | translate }}

    plat-setting-option(
    [title]="'DASHBOARD.PAYMENTS.PAYOUTS.TITLE'",
    [description]="'DASHBOARD.PAYMENTS.PAYOUTS.DESCRIPTION'",
    )
      button.button-mobile(plat-stroked-button, (click)="openPayoutMethod()")
        span {{ 'DASHBOARD.PAYMENTS.BUTTON.CHANGE' | translate }}
