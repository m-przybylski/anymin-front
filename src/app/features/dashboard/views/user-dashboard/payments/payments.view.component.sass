@import "~@anymind-ng/core/src/template/stylesheets/core/core"

.payments
	padding: 48px 0 72px
	margin: 0 auto
	max-width: 840px

	plat-setting-option
		margin: 0 0 4px 0

	&__header
		+h5
		display: block
		margin: 0 0 24px 24px

.options-container
  display: flex
  justify-content: center
  margin: 0 0 48px 0

  &__item
    width: 180px
    height: 72px
    display: flex
    justify-content: center
    align-items: center
    border-radius: 4px
    border: solid 1px $c-lines
    cursor: pointer
    transition: background-color 0.3s ease, color 0.3s ease, border-color 0.3s ease

    &:first-of-type
      margin: 0 24px 0 0

    span
      +subtitle

    &:hover
      border-color: $c-placeholder

    &.selected
      cursor: default
      border: none
      background-color: $c-text-primary

      span
        color: $c-white
