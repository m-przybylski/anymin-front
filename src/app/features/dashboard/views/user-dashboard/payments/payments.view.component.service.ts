import { Injectable } from '@angular/core';
import { PaymentsService, PutDefaultPaymentMethod } from '@anymind-ng/api';
import { Observable } from 'rxjs';

@Injectable()
export class PaymentsViewComponentService {
  constructor(private paymentsService: PaymentsService) {}

  public deletePaymentCard = (creditCardId: string): Observable<void> =>
    this.paymentsService.deleteCreditCardRoute(creditCardId);

  public setDefaultPaymentMethod = (paymentMethod: PutDefaultPaymentMethod): Observable<void> =>
    this.paymentsService.putDefaultPaymentMethodRoute(paymentMethod);
}
