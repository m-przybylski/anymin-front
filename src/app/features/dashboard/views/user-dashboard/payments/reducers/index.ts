import { ActionReducerMap, createFeatureSelector, createSelector } from '@ngrx/store';
import * as fromRoot from '@platform/reducers';
import * as fromPaymentsMethod from './payments.reducer';

export interface IPaymentsState {
  paymentsMethod: fromPaymentsMethod.IState;
}

export interface IState extends fromRoot.IState {
  payments: IPaymentsState;
}

// tslint:disable-next-line:no-any
export const reducers: ActionReducerMap<IPaymentsState, any> = {
  paymentsMethod: fromPaymentsMethod.reducer,
};

export const selectPaymentsState = createFeatureSelector<IState, IPaymentsState>('payments');

export const getPaymentsMethod = createSelector(
  selectPaymentsState,
  (state: IPaymentsState) => state.paymentsMethod,
);

export const getPaymentsMethodList = createSelector(
  getPaymentsMethod,
  fromPaymentsMethod.getPaymentsMethodList,
);
export const getPromoCodesList = createSelector(
  getPaymentsMethod,
  fromPaymentsMethod.getPromoCodesList,
);
export const getDefaultPaymentMethod = createSelector(
  getPaymentsMethod,
  fromPaymentsMethod.getDefaultPaymentMethod,
);
export const isPaymentMethodsPending = createSelector(
  getPaymentsMethod,
  fromPaymentsMethod.isPaymentMethodsPending,
);
