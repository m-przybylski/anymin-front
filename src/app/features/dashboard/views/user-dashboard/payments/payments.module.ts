import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { PaymentsViewComponent } from '@platform/features/dashboard/views/user-dashboard/payments/payments.view.component';
import { TranslateModule } from '@ngx-translate/core';
import { SharedModule } from '@platform/shared/shared.module';
import { PayoutMethodComponent } from '@platform/features/dashboard/views/user-dashboard/payments/components/payout-method/payout-method.component';
import { ModalsModule } from '@platform/shared/components/modals/modals.module';
import { BankAccountComponent } from '@platform/features/dashboard/views/user-dashboard/payments/components/payout-method/components/bank-account/bank-account.component';
import { PayPalAccountComponent } from '@platform/features/dashboard/views/user-dashboard/payments/components/payout-method/components/paypal-account/paypal-account.component';
import { InvoiceDetailsComponent } from './components/invoice-details/invoice-details.component';
import { ButtonModule, IconModule } from '@platform/shared/components/atomic-components';
import { PayoutInvoiceDetailsModule } from '@platform/shared/components/payout-invoice-details/payout-invoice-details.module';
import { AnymindComponentsModule } from '@anymind-ng/core';
import { InputsModule } from '@platform/shared/components/inputs/inputs.module';
import { MoneyDisplayModule } from '@platform/shared/components/money-display/money-display.module';
import { StoreModule } from '@ngrx/store';
import { reducers } from './reducers';
import { ContentLoaderModule } from '@platform/shared/components/content-loader/content-loader.module';

@NgModule({
  imports: [
    CommonModule,
    SharedModule,
    ModalsModule,
    RouterModule.forChild([{ path: '', component: PaymentsViewComponent }]),
    StoreModule.forFeature('payments', reducers),
    TranslateModule,
    ButtonModule,
    IconModule,
    PayoutInvoiceDetailsModule,
    AnymindComponentsModule,
    InputsModule,
    MoneyDisplayModule,
    ContentLoaderModule,
  ],
  exports: [],
  declarations: [
    PaymentsViewComponent,
    PayoutMethodComponent,
    BankAccountComponent,
    PayPalAccountComponent,
    InvoiceDetailsComponent,
  ],
  entryComponents: [PayoutMethodComponent, InvoiceDetailsComponent],
})
export class PaymentsModule {}
