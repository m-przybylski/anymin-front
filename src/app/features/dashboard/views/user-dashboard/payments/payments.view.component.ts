import { Component, OnInit } from '@angular/core';
import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { PayoutMethodComponent } from '@platform/features/dashboard/views/user-dashboard/payments/components/payout-method/payout-method.component';
import { InvoiceDetailsComponent } from './components/invoice-details/invoice-details.component';
import { AddPaymentCard } from '@platform/shared/components/modals/payments/add-payment-card/add-payment-card.component';
import { FormBuilder, FormGroup } from '@angular/forms';
import { PutDefaultPaymentMethod } from '@anymind-ng/api';
import { PaymentsViewComponentService } from '@platform/features/dashboard/views/user-dashboard/payments/payments.view.component.service';
import { ActivatedRoute } from '@angular/router';
import { AlertService, LoggerFactory } from '@anymind-ng/core';
import { Logger } from '@platform/core/logger';
import { HttpErrorResponse } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { PromoCodeComponent } from '@platform/shared/components/modals/payments/promo-code/promo-code.component';
import { select, Store } from '@ngrx/store';
import * as fromPayments from './reducers';
import { FetchPaymentsDetailsInitAction } from '@platform/features/dashboard/views/user-dashboard/payments/actions/payments-init.actions';
import { LoadPaymentsMethodOnDeleteSuccessAction } from '@platform/features/dashboard/views/user-dashboard/payments/actions/payments-api.actions';

@Component({
  selector: 'plat-payments',
  templateUrl: './payments.view.component.html',
  styleUrls: ['./payments.view.component.sass'],
  providers: [PaymentsViewComponentService],
})
export class PaymentsViewComponent extends Logger implements OnInit {
  public paymentsCardFormGroup = new FormGroup({});
  public paymentsCardControlName = 'paymentsCardControlName';
  public currentPaymentMethodId = '';
  public promoCodeList$ = this.store.pipe(select(fromPayments.getPromoCodesList));
  public paymentsCardList$ = this.store.pipe(select(fromPayments.getPaymentsMethodList));

  public isPending$ = this.store.pipe(select(fromPayments.isPaymentMethodsPending));
  private currentPromoCodeId = '';
  private currentPaymentCardId = '';

  constructor(
    private store: Store<fromPayments.IState>,
    private route: ActivatedRoute,
    private alertService: AlertService,
    private ngbModalService: NgbModal,
    private fb: FormBuilder,
    private paymentsViewComponentService: PaymentsViewComponentService,
    loggerFactory: LoggerFactory,
  ) {
    super(loggerFactory.createLoggerService('PaymentsViewComponent'));
  }

  public ngOnInit(): void {
    this.paymentsCardFormGroup = this.fb.group({
      paymentsCardControlName: '',
    });

    this.assignStoreData();
  }

  public openInvoiceModal = (): void => {
    this.ngbModalService.open(InvoiceDetailsComponent);
  };

  public openPayoutMethod = (): void => {
    this.ngbModalService.open(
      PayoutMethodComponent,
    ).componentInstance.getPayoutMethod = this.route.snapshot.data.getPayoutMethod;
  };

  public onAddPaymentCard = (): void => {
    this.ngbModalService.open(AddPaymentCard);
  };

  public onAddPromoCode = (): void => {
    this.ngbModalService.open(PromoCodeComponent);
  };

  public onSelectCard = (id: string): void => {
    const currentId = this.currentPaymentMethodId;
    this.currentPaymentCardId = id;

    this.paymentsViewComponentService
      .setDefaultPaymentMethod(this.mapCurrentPaymentMethod())
      .pipe(catchError(error => this.handleErrorOnSelect(error, currentId)))
      .subscribe(() => (this.currentPaymentMethodId = id));
  };

  public onSelectPromoCode = (id: string): void => {
    const currentId = this.currentPaymentMethodId;
    this.currentPromoCodeId = id;

    this.paymentsViewComponentService
      .setDefaultPaymentMethod(this.mapCurrentPaymentMethod(id))
      .pipe(catchError(error => this.handleErrorOnSelect(error, currentId)))
      .subscribe(() => {
        this.currentPaymentMethodId = id;
      });
  };

  public onDeleteCard = (id: string): void => {
    this.paymentsViewComponentService
      .deletePaymentCard(id)
      .pipe(
        catchError(error => {
          this.loggerService.warn('Can not delete credit payment card', error);
          this.alertService.pushDangerAlert('DASHBOARD.PAYMENTS.PAYMENTS_METHOD.CARD.DELETE.ALERT');

          return of();
        }),
      )
      .subscribe(() => {
        this.store.dispatch(new LoadPaymentsMethodOnDeleteSuccessAction());
      });
  };

  private assignStoreData = (): void => {
    this.store.dispatch(new FetchPaymentsDetailsInitAction());

    this.store.pipe(select(fromPayments.getDefaultPaymentMethod)).subscribe(currentPaymentMethod => {
      this.currentPaymentMethodId = currentPaymentMethod.promoCodeId || currentPaymentMethod.creditCardId || '';
      this.currentPromoCodeId = currentPaymentMethod.promoCodeId || '';
      this.currentPaymentCardId = currentPaymentMethod.creditCardId || '';
    });
  };

  private mapCurrentPaymentMethod = (promoCodeId?: string): PutDefaultPaymentMethod => {
    if (this.currentPaymentCardId && promoCodeId) {
      return {
        creditCardId: this.currentPaymentCardId,
        promoCodeId: this.currentPromoCodeId,
      };
    } else if (promoCodeId) {
      return {
        promoCodeId,
      };
    } else {
      return {
        creditCardId: this.currentPaymentCardId,
      };
    }
  };

  // private handleError = (error: HttpErrorResponse, msg: string): Observable<void> => {
  //   this.loggerService.warn(msg, error);
  //   this.alertService.pushDangerAlert('DASHBOARD.PAYMENTS.PAYMENTS_METHOD.CARD.SET_AS_DEFAULT.ALERT');
  //
  //   return of();
  // };

  private handleErrorOnSelect = (error: HttpErrorResponse, currentId: string): Observable<void> => {
    this.currentPaymentMethodId = currentId;
    this.loggerService.warn('Can not set default payment card', error);
    this.alertService.pushDangerAlert('DASHBOARD.PAYMENTS.PAYMENTS_METHOD.CARD.SET_AS_DEFAULT.ALERT');

    return of();
  };
}
