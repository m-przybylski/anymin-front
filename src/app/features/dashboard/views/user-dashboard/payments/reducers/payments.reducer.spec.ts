import {
  PaymentsApiActions,
  PaymentsInitActions,
} from '@platform/features/dashboard/views/user-dashboard/payments/actions';
import { IState, reducer } from '@platform/features/dashboard/views/user-dashboard/payments/reducers/payments.reducer';
import { GetCreditCard, GetDefaultPaymentMethod, GetPromoCode, MoneyDto } from '@anymind-ng/api';

describe('payments.reducer', () => {
  const initialState: IState = {
    paymentsList: [],
    defaultPaymentMethod: {},
    promoCodesList: [],
    isPending: false,
  };

  describe('undefined Action', () => {
    it('should return the default state', () => {
      const result = reducer(initialState, {} as any);
      expect(result).toMatchSnapshot(initialState);
    });
  });

  describe('FetchPaymentsDetailsInit', () => {
    it('should start FetchPaymentsDetailsInit action', () => {
      const state = {
        paymentsList: [],
        promoCodesList: [],
        defaultPaymentMethod: {},
        isPending: true,
      };

      const result = reducer(state, new PaymentsInitActions.FetchPaymentsDetailsInitAction());
      expect(result).toMatchSnapshot();
    });
  });

  describe('FetchInitPromoCodesAction', () => {
    it('should start FetchInitPromoCodesAction action', () => {
      const state = {
        paymentsList: [],
        promoCodesList: [],
        defaultPaymentMethod: {},
        isPending: false,
      };

      const result = reducer(state, new PaymentsInitActions.FetchInitPromoCodesAction());
      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadDefaultPaymentsMethodSuccessAction', () => {
    it('should start LoadDefaultPaymentsMethodSuccessAction action', () => {
      const date = new Date(0);

      const payload: ReadonlyArray<GetPromoCode> = [
        {
          id: 'string',
          token: 'string',
          amount: {
            value: 2,
            currency: 'PLN',
          },
          issuer: 'string',
          status: GetPromoCode.StatusEnum.NEW,
          expiresAt: date,
          createdAt: date,
          updatedAt: date,
        },
      ];

      const state = {
        paymentsList: [],
        promoCodesList: [
          {
            id: 'string',
            token: 'string',
            amount: {
              value: 2,
              currency: 'PLN',
            },
            issuer: 'string',
            status: GetPromoCode.StatusEnum.NEW,
            expiresAt: date,
            createdAt: date,
            updatedAt: date,
          },
        ],
        defaultPaymentMethod: {},
        isPending: false,
      };

      const result = reducer(state, new PaymentsApiActions.LoadPromoCodesSuccessAction(payload));
      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadPaymentsMethodSuccessAction', () => {
    it('should start LoadPaymentsMethodSuccessAction action', () => {
      const date = new Date(0);

      const payload: ReadonlyArray<GetCreditCard> = [
        {
          maskedNumber: 'string',
          id: 'string',
          expiryDate: 'string',
          cardType: GetCreditCard.CardTypeEnum.VISA,
          createdAt: date,
        },
      ];

      const state = {
        paymentsList: [
          {
            maskedNumber: 'string',
            id: 'string',
            expiryDate: 'string',
            cardType: GetCreditCard.CardTypeEnum.VISA,
            createdAt: date,
          },
        ],
        promoCodesList: [],
        defaultPaymentMethod: {},
        isPending: false,
      };

      const result = reducer(state, new PaymentsApiActions.LoadPaymentsMethodSuccessAction(payload));
      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadDefaultPaymentsMethodSuccessAction', () => {
    it('should start LoadDefaultPaymentsMethodSuccessAction action', () => {
      const payload: GetDefaultPaymentMethod = {
        creditCardId: 'string',
        promoCodeId: 'string',
      };

      const state = {
        paymentsList: [],
        promoCodesList: [],
        defaultPaymentMethod: {
          creditCardId: 'string',
          promoCodeId: 'string',
        },
        isPending: false,
      };

      const result = reducer(state, new PaymentsApiActions.LoadDefaultPaymentsMethodSuccessAction(payload));
      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadPaymentsMethodError', () => {
    it('should start LoadPaymentsMethodError action', () => {
      const error = 'error';

      const state = {
        paymentsList: [],
        promoCodesList: [],
        defaultPaymentMethod: {},
        isPending: false,
      };

      const result = reducer(state, new PaymentsApiActions.LoadPaymentsMethodErrorAction(error));
      expect(result).toMatchSnapshot();
    });
  });
});
