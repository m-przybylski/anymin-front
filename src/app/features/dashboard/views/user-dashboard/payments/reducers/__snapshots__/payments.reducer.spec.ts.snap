// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments.reducer FetchInitPromoCodesAction should start FetchInitPromoCodesAction action 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": false,
  "paymentsList": Array [],
  "promoCodesList": Array [],
}
`;

exports[`payments.reducer FetchPaymentsDetailsInit should start FetchPaymentsDetailsInit action 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": true,
  "paymentsList": Array [],
  "promoCodesList": Array [],
}
`;

exports[`payments.reducer LoadDefaultPaymentsMethodSuccessAction should start LoadDefaultPaymentsMethodSuccessAction action 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": false,
  "paymentsList": Array [],
  "promoCodesList": Array [
    Object {
      "amount": Object {
        "currency": "PLN",
        "value": 2,
      },
      "createdAt": 1970-01-01T00:00:00.000Z,
      "expiresAt": 1970-01-01T00:00:00.000Z,
      "id": "string",
      "issuer": "string",
      "status": "NEW",
      "token": "string",
      "updatedAt": 1970-01-01T00:00:00.000Z,
    },
  ],
}
`;

exports[`payments.reducer LoadDefaultPaymentsMethodSuccessAction should start LoadDefaultPaymentsMethodSuccessAction action 2`] = `
Object {
  "defaultPaymentMethod": Object {
    "creditCardId": "string",
    "promoCodeId": "string",
  },
  "isPending": false,
  "paymentsList": Array [],
  "promoCodesList": Array [],
}
`;

exports[`payments.reducer LoadPaymentsMethodError should start LoadPaymentsMethodError action 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": false,
  "paymentsList": Array [],
  "promoCodesList": Array [],
}
`;

exports[`payments.reducer LoadPaymentsMethodSuccessAction should start LoadPaymentsMethodSuccessAction action 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": false,
  "paymentsList": Array [
    Object {
      "cardType": "VISA",
      "createdAt": 1970-01-01T00:00:00.000Z,
      "expiryDate": "string",
      "id": "string",
      "maskedNumber": "string",
    },
  ],
  "promoCodesList": Array [],
}
`;

exports[`payments.reducer undefined Action should return the default state 1`] = `
Object {
  "defaultPaymentMethod": Object {},
  "isPending": false,
  "paymentsList": Array [],
  "promoCodesList": Array [],
}
`;
